vendor/github.com/jacobsa/gcloud/gcs/read.go

```diff
@@ -153,36 +149,31 @@ func (b *bucket) NewReader(
 	return
 }
 
-
 // Custom function made to create a new reader using Storage Client Library.
 func NewReaderSCL(
-        ctx context.Context,
-        req *ReadObjectRequest, bucketName string) (rc io.ReadCloser, err error){
-	// Create a client if there is not one already created.
-        if sclClient == nil {
-		sclClient, err = storage.NewClient(ctx)
-		if err != nil {
-			err = fmt.Errorf("Error in creating the client: %v", err)
-			return
-		}
-		fmt.Println("Client Created")
+	ctx context.Context,
+	req *ReadObjectRequest, bucketName string, sclClient *storage.Client) (rc io.ReadCloser, err error) {
+	// If client is "nil", it means that there was some problem in initializing client in newBucket function of bucket.go file.
+	if sclClient == nil {
+		err = fmt.Errorf("Error in creating client through GSCL: %v", err)
```

Comment:
same here also. err is the return variable here and by the time the code reaches here its not even initialized, why are you using err in fmt.Errorf()

Reply:
Since err is a named return parameter so I think we don't have to initialize it beforehand. It will already be initialized once the function is called.